include/meta/meta.hpp

```diff
@@ -934,13 +942,31 @@ namespace meta
     template <typename... Args>
     using transform = eval<detail::transform_<Args...>>;
 
+    ////////////////////////////////////////////////////////////////////////////
+    // filter
+    /// \cond
+    template <typename Predicate> struct filter_
+    {
+        template <typename State, typename A>
+        using apply = meta::if_<meta::apply<Predicate, A>,
+                                meta::push_back<State, A>, State>;
+    };
+    /// \endcond
+
+    /// Returns a new meta::list where all the elements of \p List A that
+    /// satisfy the Metafunction Class \p Predicate such that
+    /// `apply<Pred,A>::%value` is \c true have been removed.
+    template <typename List, typename Predicate>
+    using filter = meta::foldl<List, meta::list<>, filter_<Predicate>>;
```

Comment:
I took a look at the MPL lambda facility but it is a beast. I tried to come up with a simple way to modify/extend `apply` to allow it to handle place holders but failed.

I'll try again but I think such a feature is only worth it in meta if it turns out to be a minimal change.